R/watcher.R

```diff
@@ -12,10 +10,24 @@ watchout <- function(handler = new_output_handler(),
     dev <- dev.cur()
     defer(dev.off(dev), frame)
   }
+      
+  # Maintain a list of outputs that we'll grow over time
+  output <- list()
+  i <- 1
+  push <- function(value) {
+    output[i] <<- list(value)
+    i <<- i + 1
+    invisible()
+  }
+
+  # record whether or not we've seen an error
```

Comment:
This error handling is a little inelegant, but I'm pretty sure it'll go away in a future PR.